crates/apollo-router-core/src/naive_introspection.rs

```diff
@@ -53,7 +53,13 @@ impl NaiveIntrospection {
                 .iter()
                 .zip(responses)
                 .filter_map(|(cache_key, response)| match response.into_result() {
-                    Ok(introspection_value) => Some((cache_key.into(), introspection_value)),
+                    Ok(value) => {
+                        let mut response = Response::builder().build();
+                        response
+                            .insert_data(&Path::empty(), value)
```

Comment:
that could be a method in the response builder, like `with_data(value)`

Reply:
Good catch! Fixed in 1fbffe5 cc @o0Ignition0o (yet another small knit with naive introspection xD)